library/java/io/envoyproxy/envoymobile/engine/AndroidNetworkMonitor.java

```diff
@@ -34,6 +34,7 @@ public class AndroidNetworkMonitor extends BroadcastReceiver {
   private static volatile AndroidNetworkMonitor instance = null;
 
   private int previousNetworkType = ConnectivityManager.TYPE_DUMMY;
+  private long engineHandle;
```

Comment:
How about we store a reference to the engine, and let the engine make the JNI call, so we don't have to expose the handle?

Reply:
Giving that a shot in [22148bc](https://github.com/envoyproxy/envoy-mobile/pull/2149/commits/22148bcf9678f861fe0fc0c451c969f27fc638b6)